agents/ten_packages/extension/message_collector/src/extension.py

```diff
@@ -0,0 +1,150 @@
+
+# record the cached text data for each stream id
+cached_text_map = {}
+
+
+class MessageCollectorExtension(Extension):
+    def on_init(self, ten_env: TenEnv) -> None:
+        logger.info("MessageCollectorExtension on_init")
+        ten_env.on_init_done()
+
+    def on_start(self, ten_env: TenEnv) -> None:
+        logger.info("MessageCollectorExtension on_start")
+
+        # TODO: read properties, initialize resources
+
+        ten_env.on_start_done()
+
+    def on_stop(self, ten_env: TenEnv) -> None:
+        logger.info("MessageCollectorExtension on_stop")
+
+        # TODO: clean up resources
+
+        ten_env.on_stop_done()
+
+    def on_deinit(self, ten_env: TenEnv) -> None:
+        logger.info("MessageCollectorExtension on_deinit")
+        ten_env.on_deinit_done()
+
+    def on_cmd(self, ten_env: TenEnv, cmd: Cmd) -> None:
+        cmd_name = cmd.get_name()
+        logger.info("on_cmd name {}".format(cmd_name))
+
+        # TODO: process cmd
+
+        cmd_result = CmdResult.create(StatusCode.OK)
+        ten_env.return_result(cmd_result, cmd)
+
+    def on_data(self, ten_env: TenEnv, data: Data) -> None:
+        """
+        on_data receives data from ten graph.
+        current suppotend data:
+          - name: text_data
+            example:
+            {"name": "text_data", "properties": {"text": "hello", "is_final": true, "stream_id": 123, "end_of_segment": true}}
+        """
+        logger.info(f"on_data")
+
+        text = ""
+        final = True
+        stream_id = 0
+        end_of_segment = False
+
+        try:
+            text = data.get_property_string(TEXT_DATA_TEXT_FIELD)
+        except Exception as e:
+            logger.exception(
+                f"on_data get_property_string {TEXT_DATA_TEXT_FIELD} error: {e}"
+            )
+
+        try:
+            final = data.get_property_bool(TEXT_DATA_FINAL_FIELD)
+        except Exception as e:
+            logger.exception(
+                f"on_data get_property_bool {TEXT_DATA_FINAL_FIELD} error: {e}"
+            )
+
+        try:
+            stream_id = data.get_property_int(TEXT_DATA_STREAM_ID_FIELD)
+        except Exception as e:
+            logger.exception(
+                f"on_data get_property_int {TEXT_DATA_STREAM_ID_FIELD} error: {e}"
+            )
+
+        try:
+            end_of_segment = data.get_property_bool(TEXT_DATA_END_OF_SEGMENT_FIELD)
+        except Exception as e:
+            logger.exception(
+                f"on_data get_property_bool {TEXT_DATA_END_OF_SEGMENT_FIELD} error: {e}"
+            )
+
+        logger.debug(
+            f"on_data {TEXT_DATA_TEXT_FIELD}: {text} {TEXT_DATA_FINAL_FIELD}: {final} {TEXT_DATA_STREAM_ID_FIELD}: {stream_id} {TEXT_DATA_END_OF_SEGMENT_FIELD}: {end_of_segment}"
+        )
+
+        # We cache all final text data and append the non-final text data to the cached data
+        # until the end of the segment.
+        if end_of_segment:
+            if stream_id in cached_text_map:
```

Comment:
The LLM's stream_id will be 0 after the cmd_conversion removed, will it be properly handled by the web side?

Reply:
i searched 999 but it is not used in the entire project. so should be fine